headers/sysinfoapi/sysinfoapi.go

```diff
@@ -0,0 +1,164 @@
+type WinComputerNameFormat int
+
+// Definitions for WinComputerNameFormat constants
+const (
+	ComputerNameNetBIOS WinComputerNameFormat = iota
+	ComputerNameDNSHostname
+	ComputerNameDNSDomain
+	ComputerNameDNSFullyQualified
+	ComputerNamePhysicalNetBIOS
+	ComputerNamePhysicalDNSHostname
+	ComputerNamePhysicalDNSDomain
+	ComputerNamePhysicalDNSFullyQualified
+	ComputerNameMax
+)
+
+var (
+	kernel32                 = windows.NewLazySystemDLL("kernel32.dll")
+	procGetSystemInfo        = kernel32.NewProc("GetSystemInfo")
+	procGlobalMemoryStatusEx = kernel32.NewProc("GlobalMemoryStatusEx")
+	procGetComputerNameExW   = kernel32.NewProc("GetComputerNameExW")
+)
+
+// GlobalMemoryStatusEx retrieves information about the system's current usage of both physical and virtual memory.
+// https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-globalmemorystatusex
+func GlobalMemoryStatusEx() (MemoryStatus, error) {
+	var mse memoryStatusEx
+	mse.dwLength = (uint32)(unsafe.Sizeof(mse))
+	r1, _, err := procGlobalMemoryStatusEx.Call(uintptr(unsafe.Pointer(&mse)))
+
+	if ret := *(*bool)(unsafe.Pointer(&r1)); ret == false {
+		return MemoryStatus{}, err
+	}
+
+	return MemoryStatus{
+		MemoryLoad:           mse.DwMemoryLoad,
+		TotalPhys:            mse.UllTotalPhys,
+		AvailPhys:            mse.UllAvailPhys,
+		TotalPageFile:        mse.UllTotalPageFile,
+		AvailPageFile:        mse.UllAvailPageFile,
+		TotalVirtual:         mse.UllTotalVirtual,
+		AvailVirtual:         mse.UllAvailVirtual,
+		AvailExtendedVirtual: mse.UllAvailExtendedVirtual,
+	}, nil
+}
+
+// GetSystemInfo is an idiomatic wrapper for the GetSystemInfo function from sysinfoapi
+// https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getsysteminfo
+func GetSystemInfo() SystemInfo {
+	var info lpSystemInfo
+	procGetSystemInfo.Call(uintptr(unsafe.Pointer(&info)))
+	return SystemInfo{
+		Arch:                      ProcessorArchitecture(info.Arch.WProcessorArchitecture),
+		PageSize:                  info.DwPageSize,
+		MinimumApplicationAddress: info.LpMinimumApplicationAddress,
+		MaximumApplicationAddress: info.LpMinimumApplicationAddress,
+		ActiveProcessorMask:       info.DwActiveProcessorMask,
+		NumberOfProcessors:        info.DwNumberOfProcessors,
+		ProcessorType:             info.DwProcessorType,
+		AllocationGranularity:     info.DwAllocationGranularity,
+		ProcessorLevel:            info.WProcessorLevel,
+		ProcessorRevision:         info.WProcessorRevision,
+	}
+}
+
+// GetComputerName wraps the GetComputerNameW function in a more Go-like way
+// https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getcomputernamew
+func GetComputerName(f WinComputerNameFormat) (string, error) {
+	// 1kb buffer to accept computer name. This should be more than enough as the maximum size
+	// returned is the max length of a DNS name, which this author believes is 253 characters.
+	size := 1024
+	var buffer [4096]uint16
```

Comment:
1kb or 8kb? :)
I this as you mention in the comment that 256 (UTF-16) characters should be the limit of the how much can be returned.

Reply:
My bad, I've made size a const 1024 and used that throughout the function. Much clearer now.